mh/pat.py

```diff
@@ -2405,6 +2424,18 @@ def notify_layer_departure(self):
             ntc_data = pati.lp2_string(self.session.capcom_id)
             self.server.layer_broadcast(self.session, PatID4.NtcLayerOut,
                                         ntc_data, 0)
+        if self.session.local_info['circle_id'] is not None:
+            circle = self.session.get_circle()
+            circle_index = circle.parent.circles.index(circle) + 1
+            self.notify_circle_leave(circle_index, 0)
+
+    def notify_circle_leave(self, circle_index, seq):
+        circle = self.session.get_circle()
+        self.sendNtcCircleLeave(circle, circle_index, seq)
+        if circle.leader == self.session:
+            self.sendNtcCircleBreak(circle, seq)
```

Comment:
Add a `TODO` for handling when we are suppose to transfer the circle host

Reply:
In my work-in-progress branch for that feature, this happens in the `notify_circle_departure` function that is called before `notify_layer_departure`. There's a good chance it'll happen in a separate function. I'll add the TODO regardless, but this is something to keep in mind.